1-theory/3-api-contracts.adoc

```diff
@@ -0,0 +1,248 @@
+____
+
+Thinking about writing an API contract might sound a bit odd if you just think about an API as sharing some data or as a convenient way to trigger commands on another server, but people need to know what that data is, or what arguments they should send to those commands.
+
+The enforceable bit is important too. We can use our contracts in our test suite to confirm that our actual APIs are following the rules of the contract.
+
+NOTE: Some people use the term "API Specifications" to describe all this, but the word "specification" is used to mean a lot of different things. We just used it in the last chapter for something totally different, so let's stick to API Contracts to avoid confusion.
+
+Written down exactly from an early point (and agreed upon) means that there are far fewer surprises throughout the API lifecycle, from planning, through development, and into usage by clients.
+
+Having a good contract means that API developers can be confident that:
+
+1. The interface is doing what they intend
+2. The interface is going to be useful for client developers
+3. The interface is understood well by client developers
+4. The interface is not changing accidentally when code changes
+
+This confidence will save a lot of time and money for whoever is building this API, as a lack of confidence in those things leads to slower rollout of the initial version, loads more time spent testing subsequent deployments, and wasted developer time having loads of meetings to explain things that could have been written down and clear to everyone already.
+
+== What forms can an API Contract take?
+
+Some folks might think "That sounds like documentation", others might think "That sounds like tests". The answer is both, and more.
+
+An API contract can exist in several forms, some more useful than others.
+
+=== The Dreaded Verbal Contract
+
+A common case is the verbal contract, where the API developers discuss it with the frontend or other client application developers as they go. The backend developers write their code, the client developers write theirs, and as various endpoints or properties are conceived, they are explained somehow: literally explained out loud over the conference room table, shouted over a hail of nerf gun pellets and ping pong, or DMed over Slack chats.
+
+----
+Fred: Hey Sarah, there’s a new "fudge" field and it can be "blah" or "whatever"
+
+Sarah: Great! Thanks I’ll chuck that in now.
+----
+
+This works for a while. The API works, the app runs fine, we're making money, and we’re onto the next thing!
+
+This method has a huge number, the main being that communication is hard. You never really know if somebody understood what you meant even if they say they did. Beyond that, not having things written down means people can forget. If another client team also needs the information and they ask another client team, they are getting a copy of a copy which might be slightly wrong.
```

Comment:
```suggestion
This method has a huge number of downsides, the main being that communication is hard. You never really know if somebody understood what you meant even if they say they did. Beyond that, not having things written down means people can forget. If another client team also needs the information and they ask another client team, they are getting a copy of a copy which might be slightly wrong.
```

I feel like there should be some kind of a bridge between this paragraph and the previous sentence. Otherwise you're saying "this works" and then "this has a huge number of downsides" without a "but" in between. If that makes sense?